lib/utils/common.py

```diff
@@ -235,36 +98,391 @@ def _check_value(data={}, keys=()):
+            "namespaces",
+            (
+                "client_read_error",
+                "client_read_success",
+                "client_write_error",
+                "client_write_success",
+            ),
+        ),
+    ),
+    FeatureCheck(
+        "UDF",
+        ServiceFieldCheck("service", ("udf_read_reqs", "udf_write_reqs")),
+        NamespacesFieldCheck(
+            "namespaces",
+            ("client_udf_complete", "client_udf_error"),
+        ),
+    ),
+    FeatureCheck(
+        "Batch",
+        ServiceFieldCheck(
+            "service",
+            ("batch_initiate", "batch_index_initiate"),
+        ),
+    ),
+    FeatureCheck(
+        "Scan",
+        ServiceFieldCheck(
+            "service",
+            (
+                "tscan_initiate",
+                "basic_scans_succeeded",
+                "basic_scans_failed",
+                "aggr_scans_succeeded",
+                "aggr_scans_failed",
+                "udf_bg_scans_succeeded",
+                "udf_bg_scans_failed",
+            ),
+        ),
+        NamespacesFieldCheck(
+            "namespaces",
+            (
+                "scan_basic_complete",
+                "scan_basic_error",
+                "scan_aggr_complete",
+                "scan_aggr_error",
+                "scan_udf_bg_complete",
+                "scan_udf_bg_error",
+            ),
+        ),
+    ),
+    FeatureCheck(
+        "SIndex",
+        ServiceFieldCheck("service", ("sindex-used-bytes-memory",)),
+        NamespacesFieldCheck("namespaces", ("memory_used_sindex_bytes",)),
+    ),
+    FeatureCheck(
+        "Query",
+        ServiceFieldCheck("service", ("query_reqs", "query_success")),
+        NamespacesFieldCheck(
+            "namespaces",
+            ("query_reqs", "query_success"),
+        ),
+    ),
+    FeatureCheck(
+        "PIndex Query",
+        NamespacesFieldCheck(
+            "namespaces",
+            (
+                "pi_query_long_basic_complete",
+                "pi_query_long_basic_error",
+                "pi_query_short_basic_complete",
+                "pi_query_short_basic_error",
+                "pi_query_aggr_complete",
+                "pi_query_aggr_error",
+                "pi_query_udf_bg_complete",
+                "pi_query_udf_bg_error",
+                "pi_query_ops_bg_complete",
+                "pi_query_ops_bg_error",
+            ),
+        ),
+    ),
+    FeatureCheck(
+        "SIndex Query",
+        NamespacesFieldCheck(
+            "namespaces",
+            (
+                "si_query_long_basic_complete",
+                "si_query_long_basic_error",
+                "si_query_short_basic_complete",
+                "si_query_short_basic_error",
+                "si_query_aggr_complete",
+                "si_query_aggr_error",
+                "si_query_udf_bg_complete",
+                "si_query_udf_bg_error",
+                "si_query_ops_bg_complete",
+                "si_query_ops_bg_error",
```

Comment:
I don't think it will ever be the only query stats but there is also pi/si_query_abort